src/sql/platform/serialization/common/serializationService.ts

```diff
@@ -80,6 +104,100 @@ export class SerializationService implements ISerializationService {
 		return undefined;
 	}
 
+	public async serializeResults(serializationRequest: SerializeDataParams): Promise<azdata.SerializeDataResult> {
+		// Validate inputs
+		if (!serializationRequest) {
+			// Throwing here as this should only be a development time error
+			throw new Error('request data for serialization is missing');
+		}
+		if (!this.hasProvider()) {
+			return <azdata.SerializeDataResult>{
+				messages: localize('noSerializationProvider', "Cannot serialize data as no provider has been registered"),
+				succeeded: false
+			};
+		}
+		try {
+			// Create a new session with the provider and send initial data
+			let provider = this.providers[0].provider;
```

Comment:
should we log an error in registerProvider if a provider is already registered?

Reply:
Good question - in the future I'd like to support a "provider per save as type" so that someone else could contribute a yml writer, for example. 

In the case someone registers another formatter we should ideally do what other scenarios in VSCode do:
- If more than 1 provider is available, ask which to use
- Save that as a user setting

I will look into doing this.